config/value.go

```diff
@@ -417,7 +417,23 @@ func (cv Value) valueStruct(key string, target interface{}) (interface{}, error)
 						if fieldValue.IsNil() {
 							fieldValue.Set(reflect.New(fieldType.Elem()))
 						}
-						fieldValue.Elem().Set(reflect.ValueOf(val))
+
+						// We cannot assign reflect.ValueOf(val) to fieldValue as is, when field is a custom type
```

Comment:
instead of "custom type" I'd maybe say "user-defined named type" to align with the official documentation on assignability